bittensor/_neuron/text/advanced_server/nucleus_impl.py

```diff
@@ -87,7 +87,54 @@ def __init__(self,
         
         # -- keeps track of gradients applied
         self.backward_gradients = 0 
-        
+        self.set_fine_tuning_params()
+
+    def set_fine_tuning_params(self):
+        r''' Set to tune only the parameter of the last layer
+        '''
+        def find_last_layer(model):    
+            r''' Recursively find the last layer in a nn.ModuleList
```

Comment:
Would this work for models that have an additional head? Ex Causallm

Reply:
yea, cause it look for the last layer of transformer, and if the last layer is a transformer with multiple heads, all of the heads would required grad.